CodeEdit/Features/NavigatorArea/SourceControlNavigator/Views/Repositories/SourceControlNavigatorRepositoriesView.swift

```diff
@@ -6,30 +6,143 @@
+                                .font(.system(size: 11))
+                        }
+                    }, icon: {
+                        if item.symbolImage != nil {
+                            Image(symbol: item.symbolImage ?? "")
+                                .foregroundStyle(item.imageColor ?? .accentColor)
+                                .opacity(controlActiveState == .inactive ? 0.5 : 1)
+                        } else {
+                            Image(systemName: item.systemImage ?? "")
+                                .foregroundStyle(item.imageColor ?? .accentColor)
+                                .opacity(controlActiveState == .inactive ? 0.5 : 1)
+                        }
+                    })
+                    .padding(.leading, 2)
+                } else {
+                    Text(item.label)
+                }
             }
+            .listRowSeparator(.hidden)
         }
+        .sheet(isPresented: $showNewBranch, content: {
+            SourceControlNavigatorNewBranchView(
+                sourceControlManager: sourceControlManager,
+                fromBranch: fromBranch
+            )
+        })
+        .contextMenu(
+            forSelectionType: RepoOutlineGroupItem.self,
+            menu: { items in
+                if !items.isEmpty,
+                   items.count == 1,
+                   let item = items.first,
+                   let branch = item.branch ?? sourceControlManager.currentBranch {
+                    Button("Checkout") {
+                        Task {
+                            do {
+                                try await sourceControlManager.checkoutBranch(branch: branch)
+                            } catch {
+                                await sourceControlManager.showAlertForError(title: "Failed to checkout", error: error)
+                            }
+                        }
+                    }
+                    .disabled(item.branch == nil || sourceControlManager.currentBranch == item.branch)
+                    Divider()
+                    Button("New Branch from \"\(branch.name)\"") {
+                        showNewBranch = true
+                        fromBranch = item.branch
+                    }
+                    Divider()
+                    Button("Delete...") {
+                        Task {
+                            do {
+                                try await sourceControlManager.deleteBranch(branch: branch)
+                            } catch {
+                                await sourceControlManager.showAlertForError(title: "Failed to delete", error: error)
+                            }
+                        }
+                    }
+                    .disabled(
+                        item.branch == nil
+                        || item.branch?.isLocal == false
+                        || sourceControlManager.currentBranch == item.branch
+                    )
+                }
+            }
```

Comment:
Is there a way to handle this a little smarter? Right now this is the same menu for every item in this navigator pane. Is it possible to create two more menus for the stashed changes items and the remotes items? Maybe:
- Stashed Changes
  - Unstash Changes
  - View Stash (blank with a todo for now)
- Remotes
  - View remote (opens remote URL in browser)
  - Remove remote (blank with a todo for now)
  - Rename (blank with a todo for now)

<img width="403" alt="Screenshot 2023-11-25 at 9 11 04 PM" src="https://github.com/CodeEditApp/CodeEdit/assets/35942988/ac960990-9e9f-4e6f-b392-41e0d6e934b2">

Reply:
This is how it works in Xcode and I notice this pattern in other apps as well. I think Apple's rule here is if it is in the same list, keep the items in the context menu the same, but only light up the applicable ones for the selected item. I can see it the other way though. Why show items that you can never click? So the user knows where things are perhaps. I plan to leave it as is for the time being. I've added more items since this screenshot. It may make more sense as we build this out more.

Reply:
Here are the actions you had mentioned with my comments...

> Unstash Changes
Do you mean apply stash, or delete stash?

> View Stash (blank with a todo for now)
This is ideally what would open when the list item is clicked.

> View remote (opens remote URL in browser)
Love the idea - what if it is an ssh url or local folder though?

> Remove remote (blank with a todo for now)
I already have a "Delete..." menu item, which deletes a branch, stash entry, or remote depending on what is clicked.

> Rename (blank with a todo for now)
We can rename branches, not remotes though. We might do this later.